acl/src/main/java/org/apache/rocketmq/acl/plain/PlainPermissionChecker.java

```diff
@@ -30,36 +31,82 @@ public void check(AccessResource checkedAccess, AccessResource ownedAccess) {
         }
-        Map<String, Byte> needCheckedPermMap = checkedPlainAccess.getResourcePermMap();
-        Map<String, Byte> ownedPermMap = ownedPlainAccess.getResourcePermMap();
+        Map<String, Byte> needCheckedResourcePermMap = checkedPlainAccess.getResourcePermMap();
+        Map<String, Byte> ownedResourcePermMap = ownedPlainAccess.getResourcePermMap();
+        Map<String, Map<String, Byte>> ownedNamespacePermMap = ownedPlainAccess.getNamespacePermMap();
 
-        if (needCheckedPermMap == null) {
+        if (needCheckedResourcePermMap == null) {
             // If the needCheckedPermMap is null,then return
             return;
         }
 
-        if (ownedPermMap == null && ownedPlainAccess.isAdmin()) {
+        if (ownedResourcePermMap == null && ownedPlainAccess.isAdmin()) {
             // If the ownedPermMap is null and it is an admin user, then return
             return;
         }
 
-        for (Map.Entry<String, Byte> needCheckedEntry : needCheckedPermMap.entrySet()) {
+        for (Map.Entry<String, Byte> needCheckedEntry : needCheckedResourcePermMap.entrySet()) {
             String resource = needCheckedEntry.getKey();
             Byte neededPerm = needCheckedEntry.getValue();
             boolean isGroup = PlainAccessResource.isRetryTopic(resource);
+            boolean isResourceContainsNamespace = PlainAccessResource.isContainNamespace(resource);
 
-            if (ownedPermMap == null || !ownedPermMap.containsKey(resource)) {
-                // Check the default perm
-                byte ownedPerm = isGroup ? ownedPlainAccess.getDefaultGroupPerm() :
-                    ownedPlainAccess.getDefaultTopicPerm();
-                if (!Permission.checkPermission(neededPerm, ownedPerm)) {
+            //the resource perm that ak owned is null or doesn't contain the resource
+            if (ownedResourcePermMap == null || !ownedResourcePermMap.containsKey(resource)) {
+                //check the namespace perm and the default perm
+                if (isMatchNamespaceAndDeafultPerm(isResourceContainsNamespace, resource, ownedNamespacePermMap, isGroup, neededPerm, ownedPlainAccess)) {
+                    continue;
+                } else {
                     throw new AclException(String.format("No default permission for %s", PlainAccessResource.printStr(resource, isGroup)));
                 }
+            } else {
+                //check whether the resource perm that the ak owned is match the needed
+                if (!Permission.checkPermission(neededPerm, ownedResourcePermMap.get(resource))) {
+                    //check the namespace perm and the default perm
+                    if (isMatchNamespaceAndDeafultPerm(isResourceContainsNamespace, resource, ownedNamespacePermMap, isGroup, neededPerm, ownedPlainAccess)) {
+                        continue;
```

Comment:
It seem that we should not check namespace or default permission here, e.g.: If ak owns `DENY` permission, it is not allowed to access the resource even namespace or default permission is `PUB` or `SUB`.